src/Database/Query/CacheableQueryBuilder.php

```diff
@@ -284,12 +290,25 @@ protected function log(string $message, string $level = 'debug')
             return false;
         }
 
-        if ($this->logLevel) {
-            Log::log($this->logLevel, "[Cacheable] {$message}");
-        } else {
-            Log::log($level, "[Cacheable] {$message}");
+        // Handle dynamic log level and log channel, so we can change it on the fly
+        // for example, if log channel is set to 'some_channel', it will log to 'some_channel' channel
+        // if log channel is set to 'default', it will log to the default channel
+        if ($this->logChannel == null || $this->logChannel == 'default') {
+            $this->logChannel = Log::getDefaultDriver();
         }
 
+        // Check if the log channel exists on the driver list
+        if (! in_array($this->logChannel, array_keys(config('logging.channels')))) {
```

Comment:
Hi, as above, such checks should not be done every time. These are config and array comparisons with every log call, which is much. Such errors can be printed once and fall back to default, but not every time. If somebody alters their settings or makes false channels, fail. Wrong configs should be corrected by providing the information to the developer, and not worked around. Thanks

Reply:
Hello, thanks for checked on it, as you can see for now, if log channel is not exists system log it and change log channel to default driver, i am just wonder, if you preferred with this system or using throw an exception? please let me know, thanks

Reply:
Hey @Sleepy4k , I would always prefer "Inform and Fallback" over "Fail", therefore no Exception, as the default channel is (should) always be available. Therefore, if a wrong channel is configured, log a warning and fall back to `default`. Thanks